### DISCUSSION TITLE
Clarification needed for Blitz Zero-API

### DISCUSSION THREAD

**Discussion by cloudcompute** [blitz-js/blitz#2166] [General]
I have several questions, pl. answer

Taken from blitz web site:

>>Instead of fetching data from the backend, you import your server code directly into your components. At build time, that function import is swapped out with an auto generated HTTP API.

a. What kind of API is it that is auto-generated. Is it REST or GraphQL?

>>Somewhere else in the web site, you also said... "Zero-API" data layer abstraction that eliminates the need for REST/GraphQL

b. Is not the "eliminate-the-need-thing" contradictory to the "generated-API-thing"?

>>The generated API can also be used by apps & third-parties.

c. Does this mean that both applications written for web browsers and mobile-devices can use the same API? This feature is very important to most of the developers. Kindly explain if Blitz has any limitations in this regard.

>>About Blitz, I also read somewhere 90% of the time, we don't need APIs unless we are building a software product whose APIs we need to give to the client (user of this software product) applications.

d. APIs are very much needed if we are supposed to build both web and mobile app so that we can use the same backend code for different types of applications. Why did someone say that APIs are not needed, kindly explain.

e. A developer (user of Blitz) can write the GraphQL code i.e. mutations, query, etc. Am I right?

f. What exactly is the difference between the GraphQL functionality provided by Blitz and the normal GraphQL from the developer's point of view?

g. Last but not the least, what is the expected date of release of Blitz first stable version?

Thanks in advance.

**Comment by flybayer** [comment]
> a. What kind of API is it that is auto-generated. Is it REST or GraphQL?

Neither, it's RPC. See https://blitzjs.com/docs/rpc-specification for more details.

> b. Is not the "eliminate-the-need-thing" contradictory to the "generated-API-thing"?

What this means is that it eliminates the need for you to write and maintain the API

> c. Does this mean that both applications written for web browsers and mobile-devices can use the same API? This feature is very important to most of the developers. Kindly explain if Blitz has any limitations in this regard.

Yes, we have some users taking this approach. Other users are adding a graphql API just for their mobile app. Making mobile app integration first class will be a key focus after 1.0

> d. APIs are very much needed if we are supposed to build both web and mobile app so that we can use the same backend code for different types of applications. Why did someone say that APIs are not needed, kindly explain.

They were probably saying that many apps don't have a mobile app, but this is becoming less common. And we fully want to address this use case.

> e. A developer (user of Blitz) can write the GraphQL code i.e. mutations, query, etc. Am I right?

"Blitz queries" and "Blitz Mutations" are not GraphQL code. They just use shared naming. But you can add a graphql endpoint and use the blitz queries/mutations as your resolvers.

> f. What exactly is the difference between the GraphQL functionality provided by Blitz and the normal GraphQL from the developer's point of view?

Blitz doesn't use graphql

> g. Last but not the least, what is the expected date of release of Blitz first stable version?

May or June of this year.

**Comment by cloudcompute** [comment]
Hope the 1.0 would be coming soon. Kindly tell the expected / planned date.

**Reply by flybayer** [reply]
Q3 of this year!

**Reply by cloudcompute** [reply]
@flybayer 

Can we start developing a business app on top of blitz with the assumption that its v1.0 will be released in a month or two? Actually, what deters me is when I see the number of pending issues/bugs.

**Reply by flybayer** [reply]
I can't promise an exact date for 1.0, but yes this is a great time to start developing a business with Blitz. The vast majority of current issues are dev-time issues and will be resolve once the next.js fork migration is fully complete (should be by end of this month). 

Production runtime code is very solid with no expected changes. And there's almost no planned breaking changes before 1.0. Thousands of businesses are already using Blitz in production, many since the alpha days. So of course your risk tolerance may vary here, but almost everyone I've heard said it's worth using Blitz, even in the early days.
